src/binary/binary.c

```diff
@@ -0,0 +1,78 @@
+#include "binary.h"
+#include <stdbool.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+char* intToBin(int num)
+{
+    char* bits = calloc(32, sizeof(char));
+    for (int i = 0; i < 32; i++) {
+        bits[31 - i] = (num >> i) & 1;
+    }
+    return bits;
+}
+
+int binToInt(char* num)
+{
+    int res = 0;
+    // if num - positive
+    if (num[0] == 0) {
+        for (int i = 31; i >= 0; i--) {
+            if (num[i]) {
+                res += 1 << 31 - i;
```

Comment:
```suggestion
                res += 1 << (31 - i);
```
Если скобочки стоят так, то понимаю, почему оно работает, иначе нет. 

А если биты от 1 до 32 переводить как обычное положительное число, а потом вычесть 2^31, если число отрицательное, не будет ли код проще?